### PR TITLE
net/freeradius: add sqlite support

### PR DESCRIPTION
**Pull request by mimugmail**
Add SQLite, please only master, not stable. Need to do some testing with Captive Portal. Perhaps also @borisneubert might help with this? 

@AdSchellevis does CP already sent accouting to radius?

### CODE REVIEW

**Comment by borisneubert** [comment]
Please let me know what help you like!

**Comment by fichtner** [comment]
If you want this in the devel version of 18.7.1 let me know... about to finish the build....

**Comment by AdSchellevis** [comment]
@mimugmail yes, it sends account packages to radius using the authenticator object ( https://github.com/opnsense/core/blob/d7256733934d0b5c28e5729ea409af384f769203/src/opnsense/mvc/app/library/OPNsense/Auth/Radius.php#L278), very practical to be able to do this locally by the way.

One small note, you can't set limits at byte counters at the moment, only time based lengths are supported.

File: net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/mods-enabled-sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{% if helpers.exists('OPNsense.freeradius.general.enabled') and OPNsense.freeradius.general.enabled == '1' %}
+{%   if helpers.exists('OPNsense.freeradius.general.sqlite') and OPNsense.freeradius.general.sqlite == '1' %}
+
+sql {
+        driver = "rlm_sql_sqlite"
+        sqlite {
+                # Path to the sqlite database
+                filename = "/tmp/freeradius.db"
```
Comment (AdSchellevis) [inline net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/mods-enabled-sql]: @mimugmail I think the database itself should be outside /tmp/, the session data won't survive a reboot.

Comment (fichtner) [reply net/freeradius/src/opnsense/service/templates/OPNsense/Freeradius/mods-enabled-sql]: Something like clamav uses in the rc.conf:

clamav_clamd_var_mfs="/var/db/clamav"

**Comment by fichtner** [comment]
Ready for 18.7.1 devel ? :D

**Comment by mimugmail** [comment]
Let's start with /usr/loca/etc/raddb/ .. when possible please devel pkg with 18.7.1 so Boris must not enter 100 opnsense-patch commands :)

**Comment by mimugmail** [comment]
Ready <3

**Comment by fichtner** [comment]
Merged, thanks!

**Comment by mimugmail** [comment]
@borisneubert When you are in 18.7.1, can you Install os-freeradius-devel via CLI, enable SQLite and just let it run to see if it doesnt break your setup?

**Comment by borisneubert** [comment]
I am at 18.7.1_3. I did
`pkg install os-freeradius-devel`
Rebooted.
Enabled SQLite.
Will test tomorrow or at the weekend.

**Comment by borisneubert** [comment]
It works for me as before.

**Comment by mimugmail** [comment]
Cool! Do you see the db in /usr/local/etc/raddb growing?

**Comment by borisneubert** [comment]
I looked at the radacct table which contains all user requests seen so far since enabling the devel version. As is the case for the radpostauth table.

**Comment by borisneubert** [comment]
This has been integrated in the 18.7.3 release. How do I return from os-freeradius-devel to os-freeradius?

**Comment by mimugmail** [comment]
Just remove the devel plugin and install the stable one. It shouldnt touch your existing config (but please do a backup before)

**Comment by fichtner** [comment]
Simply installing os-freeradius from the GUI will take care of everything. :)

**Comment by borisneubert** [comment]
Franco's suggestion did it. Settings survived. Can this issue be closed?

**Comment by fichtner** [comment]
\o/ yay, glad to hear. already closed, it's a code pull request and it has been merged ❤️ 

or is there a ticket somewhere too?

**Comment by borisneubert** [comment]
Oops, I was mistaken, a pull request, indeed. No issue.

**Pull request merged**